.github/workflows/nodejs.yml

```
@@ -2,6 +2,9 @@ name: Node.js CI
 
 on: [push]
 
+permissions:
+    contents: read
```

Nitpick: it looks like we use two spaces elsewhere in this file. Could you indent it that way?

```suggestion
  contents: read
```